internal/anti_cheat.go

```diff
@@ -0,0 +1,30 @@
+package internal
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func antiCheatTest(stageHarness *test_case_harness.TestCaseHarness) error {
+	logger := stageHarness.Logger
+	executable := stageHarness.Executable
+
+	cheatExitCode := 0
+	input := "foobar"
+	pattern := "fo{1,2}bar"
+
+	executable.TimeoutInMilliseconds = 1000
+	result, err := executable.RunWithStdin([]byte(input), "-E", pattern)
+	if err != nil && err.Error() == "execution timed out" {
+		return nil
+	}
+
+	if result.ExitCode == cheatExitCode {
```

Comment:
@andy1li I worry that we'll get false positives here. What if a program is implemented poorly such that it returns 0 on unknown inputs? 

I think we'll want to run two tests - one for exit code 1 and one for exit code 0, and make sure that BOTH of those pass. If that happens, it's safe to say that the implementation actually works (i.e. anti-cheat should trigger)